src/helix/dom.cljc

```diff
@@ -161,18 +162,23 @@
 
   Use the special & or :& prop to merge dynamic props in."
   [type & args]
-  (if (map? (first args))
-    `^js/React.Element (.createElement
-                        (hx/get-react)
-                        ~type
-                        (impl.props/dom-props ~(first args))
-                        ~@(rest args))
-    `^js/React.Element (.createElement
-                        (hx/get-react)
-                        ~type
-                        nil
-                        ~@args)))
-
+  (let [?p (first args)
+        has-props? (map? ?p)
+        children* (if has-props?
+                    (rest args)
+                    args)
+        multiple-children (and children* (next children*))
+        children (if multiple-children
+                   (tl/->JSValue children*)
+                   (first children*))
+        props* (when has-props? ?p)
+        key (:key props*)
+        emit-fn (if multiple-children
+                  `hx/jsxs
+                  `hx/jsx)]
+    (if (some? key)
+      `^js/React.Element (~emit-fn ~type (impl.props/dom-props ~props* ~children) ~key)
```

Comment:
an issue I'm thinking about, not sure if it's blocking: previously, a consumer could pass in a key using dynamic props e.g.

```clojure
($ foo {& (assoc props :key "foo"})
```

however, because we need to emit the correct arity at compile time now, I'm not sure this works any longer.

I'll search through our code at work to see if this pattern is used anywhere, but I wonder if React emits a warning at all, or if we should emit a warning about it along with this change.

Reply:
oh nice, i won't worry about it then